### PR TITLE
Clarify references to RFC776 to address comments in https://github.com/huitema/dnsoquic/pull/100

### PR DESCRIPTION
**Pull request by saradickinson**
Add more specific references to RFC7766 and reuse the wording from there to address comments in https://github.com/huitema/dnsoquic/pull/100

### CODE REVIEW

File: draft-ietf-dprive-dnsoquic.md [hunk 1]
```diff
@@ -550,30 +550,32 @@ to the following recommendation:
...
 updated guidance on DNS over TCP, some of which is applicable to DoQ. This
 section attempts to specify which and how those considerations apply to DoQ.
 
 ### Connection Reuse
 
 Historic implementations of DNS clients are known to open and close TCP
-connections for each DNS query. To avoid excess QUIC connections, each with a
-single query, clients SHOULD reuse a single QUIC connection to the recursive
-resolver.
+connections for each DNS query. As outlined in section 6.2.1 of {{!RFC7766}}, to
```
Comment (martinthomson) [inline draft-ietf-dprive-dnsoquic.md:560]:
```suggestion
connections for each DNS query. As outlined in Section 6.2.1 of {{!RFC7766}}, to
```

I'd move the `{{` as well, but that can wait until Christian has updated his copy of kramdown.

**Review by martinthomson — COMMENTED** [review]
The problem with 7766 is that while some of these points are valid, some are not.

Yes, amortizing setup costs is common between TCP and DoQ.

No, making concurrent requests is not the same with TCP and DoQ.

Though there are more similarities than differences, DoQ has a different concurrency design to RFC 7766.

I would prefer to instead say that this is the same guidance as is provided in RFC 7766, but only as advisory.

File: draft-ietf-dprive-dnsoquic.md (hunk 1 above)
Comment (martinthomson) [reply draft-ietf-dprive-dnsoquic.md:560]:
```suggestion
connections for each DNS query. To
```

File: draft-ietf-dprive-dnsoquic.md [hunk 2]
```diff
@@ -550,30 +550,32 @@ to the following recommendation:
...
+amortise connection setup costs, both clients and servers SHOULD support
+connection reuse by sending multiple queries and responses over a single
+persistent QUIC connection.
 
-In order to achieve performance on par with UDP, DNS clients SHOULD send their
-queries concurrently over the QUIC streams on a QUIC connection. That is, when
-a DNS client sends multiple queries to a server over a QUIC connection, it
-SHOULD NOT wait for an outstanding reply before sending the next query.
+As outlined in Section 6.2.1.1 of {{!RFC7766}}, in order to achieve performance
```
Comment (martinthomson) [inline draft-ietf-dprive-dnsoquic.md:565]:
```suggestion
In order to achieve performance
```

File: draft-ietf-dprive-dnsoquic.md [hunk 3]
```diff
@@ -550,30 +550,32 @@ to the following recommendation:
 
 ## Connection Handling
 
-"DNS Transport over TCP - Implementation Requirements" {{?RFC7766}} provides
+"DNS Transport over TCP - Implementation Requirements" {{!RFC7766}} provides
 updated guidance on DNS over TCP, some of which is applicable to DoQ. This
 section attempts to specify which and how those considerations apply to DoQ.
```
Comment (martinthomson) [inline draft-ietf-dprive-dnsoquic.md:555]:
```suggestion
"DNS Transport over TCP - Implementation Requirements" {{!RFC7766}} provides
updated guidance on the use of DNS over TCP. This section provides similar
advice on connection handling for DoQ.
```

File: draft-ietf-dprive-dnsoquic.md [hunk 4]
```diff
@@ -550,30 +550,32 @@ to the following recommendation:
...
 ### Resource Management and Idle Timeout Values
 
 Proper management of established and idle connections is important to the
 healthy operation of a DNS server. An implementation of DoQ SHOULD follow best
-practices similar to those specified for DNS over TCP {{?RFC7766}}, in
+practices similar to those specified for DNS over TCP {{!RFC7766}}, in
 particular with regard to:
 
-* Concurrent Connections (Section 6.2.2)
+* Concurrent Connections (Section 6.2.2), updated by Section 6.4 of {{!RFC9103}}
```
Comment (martinthomson) [inline draft-ietf-dprive-dnsoquic.md:578]:
```suggestion
* Concurrent Connections ({{Section 6.2.2 of RFC7766}}, updated by {{Section 6.4 of !RFC9103}})
```

File: draft-ietf-dprive-dnsoquic.md [hunk 5]
```diff
@@ -550,30 +550,32 @@ to the following recommendation:
...
 * Security Considerations (Section 10)
```
Comment (martinthomson) [inline draft-ietf-dprive-dnsoquic.md:579]:
```suggestion
* Security Considerations ({{Section 10 of RFC7766}})
```

File: draft-ietf-dprive-dnsoquic.md [hunk 6]
```diff
@@ -550,30 +550,32 @@ to the following recommendation:
...
+As outlined in Section 6.2.1.1 of {{!RFC7766}}, in order to achieve performance
+on par with UDP, DNS clients SHOULD send their queries concurrently over the
+QUIC streams on a QUIC connection. That is, when a DNS client sends multiple
+queries to a server over a QUIC connection, it SHOULD NOT wait for an
+outstanding reply before sending the next query.
 
 ### Resource Management and Idle Timeout Values
 
 Proper management of established and idle connections is important to the
 healthy operation of a DNS server. An implementation of DoQ SHOULD follow best
```
Comment (martinthomson) [inline draft-ietf-dprive-dnsoquic.md:574]:
```suggestion
healthy operation of a DNS server.

An implementation of DoQ SHOULD follow best
```

File: draft-ietf-dprive-dnsoquic.md [hunk 7]
```diff
@@ -550,30 +550,32 @@ to the following recommendation:
...
-queries concurrently over the QUIC streams on a QUIC connection. That is, when
-a DNS client sends multiple queries to a server over a QUIC connection, it
-SHOULD NOT wait for an outstanding reply before sending the next query.
+As outlined in Section 6.2.1.1 of {{!RFC7766}}, in order to achieve performance
+on par with UDP, DNS clients SHOULD send their queries concurrently over the
+QUIC streams on a QUIC connection. That is, when a DNS client sends multiple
+queries to a server over a QUIC connection, it SHOULD NOT wait for an
+outstanding reply before sending the next query.
 
 ### Resource Management and Idle Timeout Values
```
Comment (martinthomson) [inline draft-ietf-dprive-dnsoquic.md:571]:
```suggestion
### Resource Management
```

**Review by martinthomson — COMMENTED** [review]
Lots of suggestions; I hope that they fix together properly.

This still leans heavily on 7766 for the stuff that is truly common between the two, but I think that it avoids any implication of the two protocols being the same.

**Comment by saradickinson** [comment]
@martinthomson I get what you wanted now and have applied all the changes (I'll deferred the section reference formatting into #121 as it is slightly easier...

**Pull request merged**